projects/aca-shared/src/lib/services/content-api.service.spec.ts

```diff
@@ -22,10 +22,96 @@
+import { HttpClientModule } from '@angular/common/http';
 
 describe('ContentApiService', () => {
   it('should be defined', () => {
     expect(ContentApiService).toBeDefined();
   });
+
+  describe('Info Drawer header Icon', () => {
+    let thumbnailService: ThumbnailService;
+    let service: any;
+
+    const mockNode = {
+      isFile: false,
+      createdByUser: { id: 'admin', displayName: 'Administrator' },
+      modifiedAt: new Date('2017-05-24T15:08:55.640Z'),
+      nodeType: 'cm:content',
+      content: {
+        mimeType: 'application/rtf',
+        mimeTypeName: 'Rich Text Format',
+        sizeInBytes: 14530,
+        encoding: 'UTF-8'
+      },
+      parentId: 'd124de26-6ba0-4f40-8d98-4907da2d337a',
+      createdAt: new Date('2017-05-24T15:08:55.640Z'),
+      path: {
+        name: '/Company Home/Guest Home',
+        isComplete: true,
+        elements: [
+          {
+            id: '94acfc73-7014-4475-9bd9-93a2162f0f8c',
+            name: 'Company Home'
+          },
+          { id: 'd124de26-6ba0-4f40-8d98-4907da2d337a', name: 'Guest Home' }
+        ]
+      },
+      isFolder: true,
+      modifiedByUser: { id: 'admin', displayName: 'Administrator' },
+      name: 'b_txt_file.rtf',
+      id: '70e1cc6a-6918-468a-b84a-1048093b06fd',
+      properties: { 'cm:versionLabel': '1.0', 'cm:versionType': 'MAJOR' },
+      allowableOperations: ['delete', 'update']
+    };
+
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        imports: [HttpClientModule, TranslateModule.forRoot()],
+        providers: [TranslateService]
+      });
+      thumbnailService = TestBed.inject(ThumbnailService);
+      service = TestBed.inject(ContentApiService);
+    });
+
+    function testNodeIcon(iconPath: string, isFoldeType: boolean, isFileType: boolean) {
+      spyOn(thumbnailService, 'getMimeTypeIcon').and.returnValue(iconPath);
+      mockNode.isFolder = isFoldeType;
+      mockNode.isFile = isFileType;
+      const value = service.getNodeIcon(mockNode);
+      expect(value).toContain(iconPath);
+    }
+
+    it('should resolve folder icon', () => {
+      testNodeIcon('assets/images/ft_ic_folder.svg', true, false);
+    });
+
+    it('should resolve smart folder icon', () => {
+      testNodeIcon('assets/images/ft_ic_smart_folder.svg', true, false);
+    });
+
+    it('should resolve link folder icon', () => {
+      testNodeIcon('assets/images/ft_ic_folder_shortcut_link.svg', true, false);
+    });
+
+    it('should resolve rule folder icon', () => {
+      testNodeIcon('assets/images/ft_ic_folder_rule.svg', true, false);
+    });
+
+    it('should resolve file icon for content type', () => {
+      testNodeIcon('assets/images/ft_ic_raster_image.svg', false, true);
+    });
+
+    it('should resolve fallback file icon for unknown node', () => {
+      spyOn(thumbnailService, 'getDefaultMimeTypeIcon').and.returnValue(`assets/images/ft_ic_miscellaneous.svg`);
```

Comment:
Shouldn't it use `testNodeIcon` as well?

Reply:
Done